chainerx_cc/chainerx/python/kwarg.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <string>
+#include <tuple>
+#include <type_traits>
+
+#include <pybind11/pybind11.h>
+
+namespace py = pybind11;
+
+namespace chainerx {
+namespace python {
+namespace python_internal {
+
+namespace meta {
+
+template <typename T>
+struct is_string
+    : public std::integral_constant<
+              bool,
+              std::is_same<char*, typename std::decay<T>::type>::value || std::is_same<const char*, typename std::decay<T>::type>::value> {
+};
+
+template <>
+struct is_string<std::string> : std::true_type {};
+
+template <typename... Conds>
+struct all : std::true_type {};
+
+template <typename Cond, typename... Conds>
+struct all<Cond, Conds...> : std::conditional<Cond::value, all<Conds...>, std::false_type>::type {};
+
+template <typename... Keys>
+using all_strings = all<is_string<Keys>...>;
+
+}  // namespace meta
+
+template <typename... Keys>
+auto GetKwargs(const py::kwargs& kwargs, Keys... keys) {
+    static_assert(meta::all_strings<Keys...>::value, "All keys should be of type string or char*");
```

Comment:
@niboshi Do you think we need this or this can be checked at the time of review itself?

Have checked that the below statement compiles
```
std::tie(device, dev) = GetKwargs(kwargs, "device", 1);
```